api/src/main/java/marquez/service/LineageService.java

```diff
@@ -114,15 +114,20 @@ public Lineage lineage(NodeId nodeId, int depth, boolean withRunFacets) {
     if (!datasetIds.isEmpty()) {
       datasets.addAll(this.getDatasetData(datasetIds));
     }
-    if (nodeId.isDatasetType()
-        && datasets.stream().noneMatch(n -> n.getId().equals(nodeId.asDatasetId()))) {
-      log.warn(
-          "Found jobs {} which no longer share lineage with dataset '{}' - discarding",
-          jobData.stream().map(JobData::getId).toList(),
-          nodeId.getValue());
-      return toLineageWithOrphanDataset(nodeId.asDatasetId());
-    }
 
+    if (nodeId.isDatasetType()) {
+      DatasetId datasetId = nodeId.asDatasetId();
+      DatasetData datasetData =
+          this.getDatasetData(datasetId.getNamespace().getValue(), datasetId.getName().getValue());
+
+      if (!datasetIds.contains(datasetData.getUuid())) {
+        log.warn(
+            "Found jobs {} which no longer share lineage with dataset '{}' - discarding",
+            jobData.stream().map(JobData::getId).toList(),
+            nodeId.getValue());
+        return toLineageWithOrphanDataset(nodeId.asDatasetId());
+      }
+    }
```

Comment:
Now we check if the uuid of the node and not the namespace+name

Reply:
Nice! Thanks for adding the `warn` log 💯